### DISCUSSION TITLE
Windows development and deploying on Heroku

### DISCUSSION THREAD

**Discussion by leandrocolocdk** [saleor/saleor#5765] [Q&A]
### What I am trying to achieve
... that works for me in heroku I only get
https://saleor2020.herokuapp.com/
I do not understand how to see the board and sales separately in heroku

### Steps to reproduce the problem
**1) first problem I have, is that I get (saleor-venv) C: \ nginx \ home \ manager \ saleor> python -m pip install -r requirements.txt**
Ignoring uwsgi: markers 'sys_platform! = "Win32"' don't match your environment
Collecting amqp == 2.5.2 (from -r requirements.txt (line 1))
--- that I turned it around, with all the virtual env
**2) I am lost, in development mode everything goes. now when I move the dashboard to
C: \ nginx \ home \ manager \ saleor-dashboard> npm run build**

> **saleor-dashboard@2.10.0 build C: \ nginx \ home \ manager \ saleor-dashboard**
> webpack -p

Starting type checking service ...
Using 1 worker with 2048MB memory limit
Hash: a6704e64c7702616fc3c
Version: webpack 4.43.0
Time: 136849ms
Built at: 2020-06-12 4:03:01
                                                     Asset Size Chunks Chunk Names
          ChevronDown.08f76c890fdd00a8fda4f82c34cc226d.svg 292 bytes [emitted]
     apple-touch-icon.5075c699d0e0d52314a1894a57d3de2d.png 6.45 KiB [emitted]
                         dashboard.169d14bf03ac4efc74e2.js 10.3 MiB 0 [emitted] [immutable] [big] dashboard
        favicon-16x16.5b4707491786b50c82fff3db9cbe6dc7.png 877 bytes [emitted]
        favicon-32x32.0a2dcc4bc3d372635f52acca06205b18.png 1.26 KiB [emitted]
                                                index.html 363 bytes [emitted]
     login-background.37d2649918db527e7f5c0dc49475a4ff.svg 16.3 KiB [emitted]
      logo-dark-small.21fe8b02e8be835f2fe498319d18dc18.svg 865 bytes [emitted]
            logo-dark.ac392e9dc7512230c3bf578ac9f67f38.svg 2.54 KiB [emitted]
           logo-light.d493430f320751f6bffac210ced37643.svg 1.2 KiB [emitted]
      menu-arrow-icon.a913a98d66feee5a02b53fd3b1fbe95b.svg 259 bytes [emitted]
    menu-catalog-icon.1f77dc1bc6be3723b89c4692cd2e7ff0.svg 501 bytes [emitted]
  menu-configure-icon.bd2d3cc3d7bb26f35bd6a3094b318afb.svg 2.61 KiB [emitted]
  menu-customers-icon.36532e1447af3350858044965c97d403.svg 679 bytes [emitted]
  menu-discounts-icon.bb886b229304794a64b9bfa9889e4c76.svg 706 bytes [emitted]
       menu-home-icon.b25fb1cf43ba70e4bd6d1f4125778699.svg 1.47 KiB [emitted]
     menu-orders-icon.7ce6771261234cc2a838f1145659b76d.svg 1.08 KiB [emitted]
menu-translation-icon.77a2c805576b0c0ae937fd2501c9371c.svg 2.21 KiB [emitted]
        not-found-404.a4d671be72e48d8ad9e8254297221171.svg 1.21 KiB [emitted]
           photo-icon.688aa070edcd81510ffe7b9ad5618420.svg 2.46 KiB [emitted]
   placeholder255x255.bb658a475f01d6538c482fd852ad782c.png 4.6 KiB [emitted]
    safari-pinned-tab.38f2ebe2cb94dda85a0fd1c31389e806.svg 838 bytes [emitted]
                 what.b7ae116ae14cafe6adfb9cfa119599c5.svg 1.49 KiB [emitted]
Entrypoint dashboard [big] = dashboard.169d14bf03ac4efc74e2.js
 [94] (webpack) /buildin/global.js 472 bytes {0} [built]
[133] ./node_modules/history/esm/history.js + 2 modules 30.6 KiB {0} [built]
      | 3 modules
[141] ./node_modules/@material-ui/styles/esm/index.js + 4 modules 10.9 KiB {0} [built]
      | 5 modules
[337] (webpack) /buildin/harmony-module.js 573 bytes {0} [built]
[636] ./package.json 6.35 KiB {0} [built]
[649] ./locale/ar.json 199 KiB {0} [built]
[650] ./locale/az.json 196 KiB {0} [built]
[651] ./locale/bg.json 200 KiB {0} [built]
[652] ./locale/bn.json 196 KiB {0} [built]
[653] ./locale/ca.json 196 KiB {0} [built]
[654] ./locale/cs.json 196 KiB {0} [built]
[655] ./locale/da.json 196 KiB {0} [built]
[656] ./locale/de.json 197 KiB {0} [built]
[657] ./locale/el.json 197 KiB {0} [built]
[976] ./src/index.tsx + 1445 modules 5.05 MiB {0} [built]
      | ./src/index.tsx 13.8 KiB [built]
      | ./src/config.ts 2.32 KiB [built]
      | ./src/types/globalTypes.ts 32.6 KiB [built]
      | ./src/intl.ts 6.82 KiB [built]
      | ./src/auth/errors.ts 432 bytes [built]
      | ./src/containers/AppState/index.ts 65 bytes [built]
      | ./src/hooks/useAppState.ts 228 bytes [built]
      | ./src/auth/index.tsx 1.38 KiB [built]
      | ./src/attributes/urls.ts 1.8 KiB [built]
      | ./src/navigation/urls.ts 749 bytes [built]
      | ./src/services/urls.ts 1.11 KiB [built]
      | ./src/warehouses/urls.ts 1.06 KiB [built]
      | ./src/auth/misc.ts 301 bytes [built]
      | ./src/configuration/index.tsx 8.68 KiB [built]
      | ./src/attributes/index.tsx 2.24 KiB [built]
      | + 1431 hidden modules
    + 1118 hidden modules

WARNING in asset size limit: The following asset (s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets:
  dashboard.169d14bf03ac4efc74e2.js (10.3 MiB)

WARNING in entrypoint size limit: The following entrypoint (s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  dashboard (10.3 MiB)
      dashboard.169d14bf03ac4efc74e2.js

WARNING in webpack performance recommendations:
You can limit the size of your bundles by using import () or require.ensure to lazy load some parts of your application.
For more info visit https://webpack.js.org/guides/code-splitting/

ERROR in C: /nginx/home/manager/saleor-dashboard/src/theme.ts
ERROR in C: /nginx/home/manager/saleor-dashboard/src/theme.ts (550.11):
TS2339: Property 'defaultProps' does not exist on type '(props: TextFieldProps) => Element'.

ERROR in C: /nginx/home/manager/saleor-dashboard/src/theme.ts
ERROR in C: /nginx/home/manager/saleor-dashboard/src/theme.ts (551,16):
TS2339: Property 'defaultProps' does not exist on type '(props: TextFieldProps) => Element'.

ERROR in C: /nginx/home/manager/saleor-dashboard/src/theme.ts
ERROR in C: /nginx/home/manager/saleor-dashboard/src/theme.ts (555.6):
TS2339: Property 'defaultProps' does not exist on type '(props: CardProps) => Element'.

ERROR in C: /nginx/home/manager/saleor-dashboard/src/theme.ts
ERROR in C: /nginx/home/manager/saleor-dashboard/src/theme.ts (556.11):
TS2339: Property 'defaultProps' does not exist on type '(props: CardProps) => Element'.

ERROR in C: /nginx/home/manager/saleor-dashboard/src/theme.ts
ERROR in C: /nginx/home/manager/saleor-dashboard/src/theme.ts (560.12):
TS2339: Property 'defaultProps' does not exist on type 'OverridableComponent <TypographyTypeMap <{}, "span" >>'.

ERROR in C: /nginx/home/manager/saleor-dashboard/src/utils/sort.ts
ERROR in C: /nginx/home/manager/saleor-dashboard/src/utils/sort.ts (52.3):
TS2322: Type 'TParams & {asc: boolean; sort: keyof TFields; } 'is not assignable to type' TParams & Partial <{asc: boolean; sort: string; }> '.
  Type 'TParams & {asc: boolean; sort: keyof TFields; } 'is not assignable to type' Partial <{asc: boolean; sort: string; }> '.
    Types of property 'sort' are incompatible.
      Type 'keyof TFields' is not assignable to type 'string'.
        Type 'string | number | symbol 'is not assignable to type' string '.
          Type 'number' is not assignable to type 'string'.
Child html-webpack-plugin for "index.html":
     1 asset
    Entrypoint undefined = index.html
    [0] ./node_modules/html-webpack-plugin/lib/loader.js!./src/index.html 466 bytes {0} [built]
    [2] (webpack) /buildin/global.js 472 bytes {0} [built]
    [3] (webpack) /buildin/module.js 497 bytes {0} [built]
        + 1 hidden module
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! saleor-dashboard@2.10.0 build: `webpack -p`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the saleor-dashboard@2.10.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR! C: \ Users \ MSI \ AppData \ Roaming \ npm-cache \ _logs \ 2020-06-12T07_03_02_073Z-debug.log

**3) heroku's answer**

**C: \ nginx \ home \ manager \ saleor> git push heroku master**
Enumerating objects: 7, done.
Counting objects: 100% (7/7), done.
Delta compression using up to 8 threads
Compressing objects: 100% (4/4), done.
Writing objects: 100% (4/4), 371 bytes | 371.00 KiB / s, donate.
Total 4 (delta 3), reused 0 (delta 0), pack-reused 0
remote: Compressing source files ... done.
remote: Building source:
remote:
remote: -----> Node.js app detected
remote:
remote: -----> Creating runtime environment
remote:
remote: NPM_CONFIG_LOGLEVEL = error
remote: NPM_CONFIG_PRODUCTION = false
remote: NODE_ENV = production
remote: NODE_MODULES_CACHE = false
remote: NODE_VERBOSE = false
remote:
remote: -----> Installing binaries
remote: engines.node (package.json): unspecified
remote: engines.npm (package.json): unspecified (use default)
remote:
remote: Resolving node version 12.x ...
remote: Downloading and installing node 12.18.0 ...
remote: Using default npm version: 6.14.4
remote:
remote: -----> Restoring cache
remote: Caching has been disabled because NODE_MODULES_CACHE = false
remote:
remote: -----> Installing dependencies
remote: Installing node modules
remote:
remote:> core-js@2.6.11 postinstall / tmp / build_f6f213c722d761fcb3461880e20e4503 / node_modules / core-js
remote:> node -e "try {require ('./ postinstall')} catch (e) {}"
remote:
remote: added 223 packages in 4.423s
remote:
remote: -----> Build
remote: Running heroku-postbuild
remote:
remote:> saleor@0.0.0 heroku-postbuild / tmp / build_f6f213c722d761fcb3461880e20e4503
remote:> npm run build-emails
remote:
remote:
remote:> saleor@0.0.0 build-emails / tmp / build_f6f213c722d761fcb3461880e20e4503
remote:> mjml --config.beautify false -l skip "templates / templated_email / source / *. mjml" -o templates / templated_email / compiled
remote:
remote:
remote: -----> Pruning devDependencies
remote: Skipping because NPM_CONFIG_PRODUCTION is 'false'
remote:
remote: -----> Build succeeded!
remote: -----> Python app detected
remote: -----> No change in requirements detected, installing from cache
remote: -----> Installing SQLite3
remote: -----> Installing requirements with pip
remote: ignoring python-magic-bin: markers 'sys_platform == "win32"' don't match your environment
remote: -----> Discovering process types
remote: Procfile declares types -> celeryworker, release, web
remote:
remote: -----> Compressing ...
remote: Done: 174.8M
remote: -----> Launching ...
remote:! Release command declared: this new release will not be available until the command succeeds.
remote: Released v28
remote: https://saleor2020.herokuapp.com/ deployed to Heroku
remote:
remote: Verifying deploy ... done.
remote: Running release command ...
remote:
remote: /app/.heroku/python/lib/python3.8/site-packages/mpmath/ctx_mp_python.py:892: SyntaxWarning: "is" with a literal. Did you mean "=="?
remote: if other is 0:
remote: /app/.heroku/python/lib/python3.8/site-packages/mpmath/ctx_mp_python.py:986: SyntaxWarning: "is" with a literal. Did you mean "=="?
remote: if other is 0:
remote: /app/saleor/urls.py:29: UserWarning: The debug toolbar was not installed. Ignore the error. settings.py should already have warned the user about it.
remote: warnings.warn (
remote: Operations to perform:
remote: Apply all migrations: account, app, auth, checkout, contenttypes, core, discount, django_prices_openexchangerates, django_prices_vatlayer, giftcard, menu, order, page, payment, plugins, product, shipping, site, sites, warehouse, webhook, wishlist
remote: Running migrations:
remote: No migrations to apply.
remote: Waiting for release .... done.
To https://git.heroku.com/saleor2020.git
   b162cae..abd4af9 master -> master

### What I expected to happen
... I know, I went everywhere and I'm pretty lost, I don't know where to go anymore, I followed all the guidelines ...
I do not know what to do. I thought about uwsgi, nginx, but I don't know how to handle all that. I downloaded cygwin to install the uwsgi, and there I can no longer put virtualenv.

### Screenshots
<! - If applicable, please add screenshots to help explain your problem. ->

**System information**
Operating system: windows 10, python 3.8.3

**Comment by eaglezero** [comment]
did you found the solution?

**Comment by leandrocolocdk** [comment]
No

Obtener Outlook para Android<https://aka.ms/ghei36>

________________________________

**Reply by karolkielecki** [reply]
Hi there! Did you follow our guide: https://docs.saleor.io/docs/developer/running-saleor/heroku. There is also information about Storefront and Dashboard.

**Comment by krzysztofwolski** [comment]
For local development please use WSL2 - it's linux subsystem. Works great on Win 10 (home edition too).
